modules/currency-l0/src/main/scala/io/constellationnetwork/currency/l0/StateChannel.scala

```diff
@@ -63,6 +65,12 @@ object StateChannel {
                 ) >> Applicative[F].pure(nextSnapshots.asLeft[Unit])
             }
         }
+        .evalMap { globalSnapshot =>
+          dataApplicationService match {
+            case Some(service) => service.onGlobalSnapshotPull(globalSnapshot)
+            case None          => Applicative[F].unit
+          }
+        }
```

Comment:
I think it would make more sense to call `onGlobalSnapshotPull` from tailRecM above, so you can pass the context (GlobalSnapshotInfo)